commit a95cf15096a9496108385bc1a47bb7d0c7d604ad (refs/changes/05/505/2)
Author: Pedro P. Candel <pedro@joyent.com>
Date:   2016-09-20T19:02:15+02:00 (3 years, 1 month ago)
    
    TOOLS-1549 sdcadm post-setup manatee fails due to missing image version
    Reviewed by: Trent Mick <trent.mick@joyent.com>

diff --git a/lib/post-setup/ha-manatee.js b/lib/post-setup/ha-manatee.js
index 812cce1..a06b042 100644
--- a/lib/post-setup/ha-manatee.js
+++ b/lib/post-setup/ha-manatee.js
@@ -290,6 +290,7 @@ function do_ha_manatee(subcmd, opts, args, cb) {
                     next(err);
                 } else {
                     img = im;
+                    arg.change.image = img;
                     next();
                 }
             });
@@ -485,7 +486,7 @@ function do_ha_manatee(subcmd, opts, args, cb) {
                     server: opts.servers[0],
                     type: 'create',
                     service: 'manatee',
-                    image: img.uuid,
+                    image: img,
                     inst: {
                         instance: newId,
                         zonename: newId,
@@ -494,6 +495,7 @@ function do_ha_manatee(subcmd, opts, args, cb) {
                         server: opts.servers[0],
                         service: 'manatee',
                         image: img.uuid,
+                        version: img.version,
                         type: 'vm'
                     }
                 },
diff --git a/lib/procedures/update-manatee-v2.js b/lib/procedures/update-manatee-v2.js
index 5d3d34a..38b5f67 100644
--- a/lib/procedures/update-manatee-v2.js
+++ b/lib/procedures/update-manatee-v2.js
@@ -863,7 +863,7 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                         server: arg.shard.async.server_uuid,
                         type: 'create',
                         service: 'manatee',
-                        image: arg.change.image.uuid,
+                        image: arg.change.image,
                         inst: {
                             instance: arg.shard.async.zoneId,
                             zonename: arg.shard.async.zoneId,
@@ -871,6 +871,7 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                             server: arg.shard.async.server_uuid,
                             service: 'manatee',
                             image: arg.change.image.uuid,
+                            version: arg.change.image.version,
                             type: 'vm'
                         }
                     },
@@ -935,7 +936,7 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                         server: arg.shard.sync.server_uuid,
                         type: 'create',
                         service: 'manatee',
-                        image: arg.change.image.uuid,
+                        image: arg.change.image,
                         inst: {
                             instance: arg.shard.sync.zoneId,
                             zonename: arg.shard.sync.zoneId,
@@ -943,6 +944,7 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                             server: arg.shard.sync.server_uuid,
                             service: 'manatee',
                             image: arg.change.image.uuid,
+                            version: arg.change.image.version,
                             type: 'vm'
                         }
                     },
@@ -995,7 +997,7 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                         server: arg.shard.primary.server_uuid,
                         type: 'create',
                         service: 'manatee',
-                        image: arg.change.image.uuid,
+                        image: arg.change.image,
                         inst: {
                             instance: arg.shard.primary.zoneId,
                             zonename: arg.shard.primary.zoneId,
@@ -1003,6 +1005,7 @@ UpdateManateeV2.prototype.execute = function manateev2Execute(opts, cb) {
                             server: arg.shard.primary.server_uuid,
                             service: 'manatee',
                             image: arg.change.image.uuid,
+                            version: arg.change.image.version,
                             type: 'vm'
                         }
                     },
